### PR TITLE
Reverted changes meant for 2.0.0 and restored Java8 compatibility

### PR DESCRIPTION
**Pull request by Lonzak**
1) Reverted be388653bc948ebfc707be58192da61eec71abd1 to maintain compatibility with iText. This kind of changes will be done in 2.0.0 onwards.

2) Commit 63a678d6089bbaaf53e0a82e6061e2de36cf9206  broke java8 compatibility

### CODE REVIEW

**Comment by RadekWikturna** [comment]
Can you please fix the problems reported by Codacy?

File: openpdf/src/main/java/com/lowagie/text/Paragraph.java [hunk 1]
```diff
@@ -204,8 +204,8 @@ public Paragraph(Phrase phrase) {
             setIndentationLeft(p.getIndentationLeft());
             setIndentationRight(p.getIndentationRight());
             setFirstLineIndent(p.getFirstLineIndent());
-            setSpacingAfter(p.getSpacingAfter());
-            setSpacingBefore(p.getSpacingBefore());
+            setSpacingAfter(p.spacingAfter());
+            setSpacingBefore(p.spacingBefore());
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/Paragraph.java:208]: This was not needed. Just marking here

File: openpdf/src/main/java/com/lowagie/text/StandardFonts.java [hunk 2]
```diff
@@ -8,37 +8,116 @@ public enum StandardFonts {
...
+    /**
+     * @deprecated Use Liberation
+     */
+    @Deprecated
+    LIBERATION_MONO_BOLD("font-fallback/LiberationMono-Bold.ttf"),
+    /**
+     * @deprecated Use Liberation
+     */
+    @Deprecated
+    LIBERATION_MONO_BOLDITALIC("font-fallback/LiberationMono-BoldItalic.ttf"),
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/StandardFonts.java:31]: Liberation-Fonts were added in OpenPDF, so these can be removed.

File: openpdf/src/main/java/com/lowagie/text/pdf/FdfReader.java [hunk 3]
```diff
@@ -114,7 +114,7 @@ protected void readPdf() throws IOException {
     protected void kidNode(PdfDictionary merged, String name) {
         PdfArray kids = merged.getAsArray(PdfName.KIDS);
         if (kids == null || kids.isEmpty()) {
-            if (!name.isEmpty())
+            if (name.length() > 0)
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/pdf/FdfReader.java:117]: Revert not needed here. Marking.

File: openpdf/src/main/java/com/lowagie/text/pdf/MappedRandomAccessFile.java [hunk 4]
```diff
@@ -244,22 +243,24 @@ private static boolean cleanOldsJDK(final java.nio.ByteBuffer buffer) {
 
 
     private static boolean cleanJava9(final java.nio.ByteBuffer buffer) {
-        return AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
-            Boolean success = Boolean.FALSE;
-            try {
+        Boolean b = AccessController.doPrivileged((PrivilegedAction<Boolean>) () -> {
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/pdf/MappedRandomAccessFile.java:246]: Could stay like that. Marking.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfCopyFieldsImp.java [hunk 5]
```diff
@@ -168,7 +168,7 @@ protected void updateCalculationOrder(PdfReader reader) {
         if (acro == null)
             return;
         PdfArray co = acro.getAsArray(PdfName.CO);
-        if (co == null || co.isEmpty())
+        if (co == null || co.size() == 0)
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/pdf/PdfCopyFieldsImp.java:171]: Marking.

File: openpdf/src/main/java/com/lowagie/text/pdf/hyphenation/SimplePatternParser.java [hunk 6]
```diff
@@ -101,6 +102,11 @@ public void parse(InputStream stream, PatternConsumer consumer) {
             SimpleXMLParser.parse(this, stream);
         } catch (IOException e) {
             throw new ExceptionConverter(e);
+        } finally {
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/pdf/hyphenation/SimplePatternParser.java:105]: Not sure if this is needed. closing a Stream in a Method which didn't open it, is not really a good think.

File: openpdf/src/test/java/com/lowagie/text/pdf/SimpleBookmarkTest.java [hunk 7]
```diff
@@ -15,7 +15,7 @@ class SimpleBookmarkTest {
     void testGetBookmarkWithNoTitle() throws IOException {
         InputStream is = getClass().getResourceAsStream("/OutlineUriActionWithNoTitle.pdf");
         PdfReader reader = new PdfReader(is);
-        List<?> list = SimpleBookmark.getBookmarkList(reader);
+        List<?> list = SimpleBookmark.getBookmark(reader);
```
Comment (asturio) [inline openpdf/src/test/java/com/lowagie/text/pdf/SimpleBookmarkTest.java:18]: Usage of not-deprecated methods inside OpenPDF should be ok.

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java [hunk 8]
```diff
@@ -506,10 +506,10 @@ public int compareTo(final PdfCrossReference reference) {
              */
             @Override
             public boolean equals(Object obj) {
-                if (!(obj instanceof PdfCrossReference other)) {
+                if (!(obj instanceof PdfCrossReference)) {
                     return false;
                 }
-
+                final PdfCrossReference other = (PdfCrossReference)obj;
```
Comment (asturio) [inline openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java:512]: Good one here!

**Review by asturio — APPROVED** [review]
Hi @Lonzak ,

if you have time take a look at my marking. If not, I can merge this and try to check if these are really ok, in a "cleanup" commit.

**Comment by asturio** [comment]
Looking tomorrow again, and merging.

**Comment by Lonzak** [comment]
I have to admit I didn't check the changes from the commit  ( be388653bc948ebfc707be58192da61eec71abd1) in detail - I just used the revert function of my git tool...
Parts of the changes like using `isEmpty()` and so on is of course better. And I would also remove the ITXT variable from PdfName...

File: openpdf/src/main/java/com/lowagie/text/Paragraph.java (hunk 1 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/Paragraph.java]: Yes you are right - I reverted it.

File: openpdf/src/main/java/com/lowagie/text/StandardFonts.java (hunk 2 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/StandardFonts.java]: Ah ok didn't know. Reverted.

File: openpdf/src/main/java/com/lowagie/text/pdf/FdfReader.java (hunk 3 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/pdf/FdfReader.java]: Reverted it

File: openpdf/src/main/java/com/lowagie/text/pdf/MappedRandomAccessFile.java (hunk 4 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/pdf/MappedRandomAccessFile.java]: reverted it

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfCopyFieldsImp.java (hunk 5 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfCopyFieldsImp.java]: reverted it

File: openpdf/src/test/java/com/lowagie/text/pdf/SimpleBookmarkTest.java (hunk 7 above)
Comment (Lonzak) [reply openpdf/src/test/java/com/lowagie/text/pdf/SimpleBookmarkTest.java]: reverted it

File: openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java (hunk 8 above)
Comment (Lonzak) [reply openpdf/src/main/java/com/lowagie/text/pdf/PdfWriter.java:512]: left it like it is...

**Comment by asturio** [comment]
The code is not compiling. I think the given test class was removed

**Comment by asturio** [comment]
I think for this PR the Codacy could be ignored.

**Pull request merged**